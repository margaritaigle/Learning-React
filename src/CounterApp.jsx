import { useState } from 'react';
import PropTypes from 'prop-types';


export const CounterApp = ( {value} ) => {

    const [ counter, setCounter ] = useState( value ); // desestructuracion del retorno de la funcion useState
    
    const handleAdd = () => {
        // setCounter( counter + 1 ); ambas son validas
        setCounter( (c) => c + 1 ); // c es el valor nuevo que va tomando el counter 
    } 

    const handleSubst = () => {
        // setCounter( counter - 1 );
        setCounter( (c) => c - 1 );
    }

    const handleReset = () => {
        setCounter( value );
    }
    
    return (
        <>
            <h1>CounterApp</h1>
            <h2> { counter } </h2>

            <button aria-label='btn-increment' 
                    onClick={handleAdd}>
                +1
            </button>
            <button aria-label='btn-decrement' 
                    onClick={handleSubst}>
                -1
            </button>
            <button aria-label='btn-reset' 
                    onClick={handleReset}>
                Reset
            </button>
        </>
    );
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired,
}


// onClick={ handleAdd } === onClick={ (event) => handleAdd(event) }