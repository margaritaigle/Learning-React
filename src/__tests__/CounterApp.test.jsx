import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { CounterApp } from "../CounterApp.jsx";

describe('tests in Counter App', () => {
    
    const value = 100;

    test('debe de hacer match con el snapshot', () => {

        const {container} = render(<CounterApp value={value} />);

        expect(container).toMatchSnapshot();
    });

    test('should show the initial value 100', () => { 
        render(<CounterApp value={value} />);

        expect( screen.getByRole('heading', { level: 2 }).innerHTML ).toContain('100');
        expect( screen.getByText(100)).toBeTruthy();
     });

     test('the button +1 should increment', () => { 
 
        render(<CounterApp value={value} />);
        userEvent.click( screen.getByText('+1') );
        
        expect( screen.getByText('101') ).toBeTruthy();
      });
     
      test('the button -1 should decrement', () => { 
  
        render(<CounterApp value={value} />);
        userEvent.click( screen.getByText('-1') );
        
        expect( screen.getByText('99') ).toBeTruthy();
      });
      
      test('the button reset must turn the value to its initial state', () => { 
  
        render(<CounterApp value={value} />);
        userEvent.click( screen.getByText('-1') );
        // userEvent.click( screen.getByText('Reset') );
        userEvent.click( screen.getByRole('button', { name: 'btn-reset' }));

        expect( screen.getByText('100') ).toBeTruthy();
      });

    });