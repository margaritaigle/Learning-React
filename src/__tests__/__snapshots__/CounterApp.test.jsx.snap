// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests in Counter App debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
     
    100
     
  </h2>
  <button
    aria-label="btn-increment"
  >
    +1
  </button>
  <button
    aria-label="btn-decrement"
  >
    -1
  </button>
  <button
    aria-label="btn-reset"
  >
    Reset
  </button>
</div>
`;
